Swamp Sneak/Assets/Scripts/rkScripts/GameControl.cs

```diff
@@ -2,38 +2,54 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.SceneManagement;
+using System;
 using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
 using UnityEditor;
 
 public class GameControl : MonoBehaviour
 {
+	// Use singleton design pattern. This variable can be accessed anywhere.
+	public static GameControl control;
+
     //Public Constants and Variables
     public const int LOADINGSCENE = 0;
     public string startScene = "start";
+	public int playerStealth = 1;
+	public bool gameWin = false;
+	public string save_file_location = "Assets/SaveData/";
+	public string save_file_name = "SaveFile.dat";
+	public PersistentData persistent_save_data;
+
     //Private Variables
-    private static bool isAwake = false;
+	private static Player player_character;		// Reference to player character initialized in Awake()
     private bool isCurScnEditable = true;
     private bool loadScn = false;
     private string currentScene, prevScene;
-
-
-    private GameObject winBox;
-
-    public int playerStealth = 1;
-    public bool gameWin = false;
+	private GameObject winBox;    
     private int level = 0;
 
-
-    //Awake is call 0
+    // Awake is call 0.
+	// Use singleton design pattern, so there will only ever be one GameControl.
+	// The GameControl can now be placed in every single scene, and the order in 
+	// which the scenes are loaded does not matter.
     private void Awake()
     {
-#pragma warning disable CS0665 // Assignment in conditional expression is always constant
-        if (isAwake = true)
-#pragma warning restore CS0665 // Assignment in conditional expression is always constant
-        {
-            DontDestroyOnLoad(this.gameObject);
-            Debug.Log("The " + this.gameObject.ToString() + " is Awake and Will Not Be Destroyed");
-        }
+		// If control does not exist, initialize it
+		if (control == null) {
+			DontDestroyOnLoad (gameObject);
+			control = this;
+			player_character = GameObject.FindWithTag ("Player").GetComponent<Player>();
+
+			Debug.Log("The " + this.gameObject.ToString() + " is Awake and Will Not Be Destroyed");
+		} 
+		// If it does already exist, and it is not this, delete it (there can only be one!)
+		else if (control != this) {
+			Destroy (gameObject);
```

Comment:
Not that Destroy will not actually exit this function. The actually script destruction will occur [after the current Update loop](https://docs.unity3d.com/ScriptReference/Object.Destroy.html). Therefore, that `LoadData` method will be called, only for this object to be destroyed. Maybe you'd want to return here?

Reply:
Good idea!!